### PR TITLE
Updates Rush, PNPM versions.

### PR DESCRIPTION
**Pull request by smurrayatwork**
This PR just updates Rush and PNPM to their latest stable versions. It also bumps the allowed NodeJS version to be between 12.13 and 14.x.

### CODE REVIEW

File: common/scripts/install-run.js [hunk 1]
```diff
@@ -1,7 +1,27 @@
 "use strict";
 // Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
 // See the @microsoft/rush package's LICENSE file for license information.
+var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
```
Comment (clairesunstudio) [inline common/scripts/install-run.js:4]: Are these new lines auto-generated from the new version?

File: rush.json [hunk 2]
```diff
@@ -121,7 +121,7 @@
    * Specify a SemVer range to ensure developers use a Node.js version that is appropriate
    * for your repo.
    */
-  "nodeSupportedVersionRange": ">=12.13.0 <13.0.0",
+  "nodeSupportedVersionRange": ">=12.13.0 <15.0.0",
```
Comment (clairesunstudio) [inline rush.json:124]: validated on node v12.18.0
Did you test if v14 worked? my zsh export PATH is having issues locating rush on v14 `command not found: rush`

**Review by clairesunstudio — APPROVED** [review]

File: common/scripts/install-run.js (hunk 1 above)
Comment (smurrayatwork) [reply common/scripts/install-run.js:4]: Yes

File: rush.json (hunk 2 above)
Comment (smurrayatwork) [reply rush.json:124]: Currently we can't, but in the future we can.

**Pull request merged**